### PR TITLE
simplify extension build and dependencies

### LINKED ISSUE #168: Remove gulp-browserify
+ gulp-browserify is a deprecated module.

A few questions before I send a PR regarding this.

+ Since we use only latest chrome, we could simply use es6 code in browser - why transpile ? (edit: no uglify for es6 )
+ And why browserify and rollup ?, if we have all our files as ES6 modules (transpile only es6 module stuff for node), it would be possible to use rollup along with rollup's commonjs, node-resolve, and json plugins. Or only have browserify and remove rollup ?

Would be happy to send a PR.

### LINKED ISSUE #224: Centralize npmlog init
Move
```
const log = (typeof process !== 'undefined' && 'version' in process) ?
    require('npmlog').log : console.log.bind(console);
```
into a central place so it can just be `require('helpers/log')` or whatever wherever it's needed

### PR DESCRIPTION
**Pull request by brendankenny**
fixes #168
fixes #224

Removes a bunch of dependencies we barely or don't use, including deprecated `gulp-browserify`. On my local machine, cold cache `npm install` of the extension goes from 101 seconds to 34 seconds, warm cache time goes from 56 seconds to 24 seconds.

### CODE REVIEW

File: extension/.babelrc [hunk 1]
```diff
@@ -1,3 +0,0 @@
-{
```
Comment (brendankenny) [inline extension/.babelrc]: thanks to Mr. @paulirish for catching this

Comment (paulirish) [inline extension/.babelrc]: 🎩

File: src/lib/drivers/extension.js [hunk 2]
```diff
@@ -17,7 +17,7 @@
 'use strict';
 
 const Driver = require('./driver.js');
-const log = (typeof process !== 'undefined' && 'version' in process) ?
+const log = (process && (!process.browser || !process.versions.electron)) ?
```
Comment (paulirish) [inline src/lib/drivers/extension.js]: I think it's time to bust this into a helper. You know you want to.

Comment (brendankenny) [inline src/lib/drivers/extension.js]: done

**Comment by paulirish** [comment]
she's a beauty. nice.

**Pull request merged**